.github/workflows/docker-build-publish.yaml

```diff
@@ -24,6 +24,9 @@ on:
       DOCKER_REGISTRY_REF:
         description: "docker registry reference of the container image"
         value: ${{ jobs.docker-env.outputs.DOCKER_REGISTRY_REF }}
+      DOCKER_REGISTRY_PUSH:
```

Comment:
@WindzCUHK I cannot find any necessity of adding a new environment variable just for the boolean flag.
`${{ github.event_name != 'pull_request' }}` looks more straightforward for the pipeline and adding a new variable `DOCKER_REGISTRY_PUSH` looks redundant.

Reply:
👀

Reply:
@ctyano
I want to push the container image as `pr-1` in the PR during the development phrase.
As there are multiple steps using the `${{ github.event_name != 'pull_request' }}` condition, it is easier to just change the variable's value.

Reply:
@WindzCUHK Do you mean that all the steps using `${{ github.event_name != 'pull_request' }}` is not final to diverge with a condition if it's a PR or not? And do we actually have any plans to change them in future?
I was wondering the benefit of switching every part as a single variable.

Reply:
@ctyano 
> Do you mean that all the steps using ${{ github.event_name != 'pull_request' }} are not final to diverge with a condition if it's a PR or not?

Yes, those steps just depends on whether the container image is pushed or not.
However, there are multiple factors to determine whether the pipeline should/can push or not.

| trigger event | PR src => dest   | push dest     | push |
|---------------|------------------|---------------|------|
| tag           |                  | docker/athenz | ✅    |
| pull_request  | athenz => athenz | docker/athenz | ❌    |
| pull_request  | fork => athenz   | docker/athenz | ❌    |
| pull_request  | fork => athenz   | ghcr/fork     | ✅    |
| pull_request  | fork => fork     | ghcr/fork     | ✅    |

> And do we actually have any plans to change them in future?

Currently, no but I think we should reconsider what should we support in the fork use cases, or just to ignore fork use cases from the pipeline.

Reply:
All PRs should not push to the registry.